src/permission/fs_permission.cc

```diff
@@ -0,0 +1,196 @@
+#include "fs_permission.h"
+#include "base_object-inl.h"
+#include "v8.h"
+
+#include <fcntl.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <filesystem>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace node {
+
+namespace permission {
+
+std::string WildcardIfDir(std::string res) {
+  std::filesystem::path path(res);
+  if (std::filesystem::is_directory(path)) {
```

Comment:
Which libuv function do you refer to? Anyway, I've updated the code to use the std::error_code approach f02bc6eb20.

Reply:
I was referring to using `uv_fs_stat()` and then checking if `stat_buf->st_mode & S_IFDIR` is non-zero. While this approach also looks good, I wonder if this could be backported to all the release lines you want to target because std::filesystem is a C++17 thing.

Reply:
The problem with `uv_fs_stat()` is that it requires a real file/folder. Sometimes, this function will be called for future folders, such as `/home/user/future-folder/` which doesn't exist _yet_, so it will throw an exception.

Reply:
From what I can tell, the current function would add an asterisk to the path only if the path points to an existing file which is a directory. I think you could implement the same thing with uv_fs_stat without throwing any exceptions in a backportable way:
```cc
  uv_fs_t req;
  int rc = uv_fs_stat(nullptr, &req, res.c_str(), nullptr);
  if (rc == 0) {
    const uv_stat_t* const s = static_cast<const uv_stat_t*>(req.ptr);
    rc = !!(s->st_mode & S_IFDIR);
  }
  uv_fs_req_cleanup(&req);
  if (rc) {
    // it is a directory, so add wildcard
  }
  // return modified path
```

Reply:
Oh, got it! Thanks for the sample code, I'll update!